bin/github-review.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+Verify a Safe singleton factory deployment PR
+
+This tool wraps the NPM script 'verify' and:
+1. Parses the RPC URL from a GitHub PR and referenced issue
+2. Automatically approves the GitHub PR
+
+USAGE
+    github-review.sh PR
+
+ARGUMENTS
+    PR          The GitHub PR number to review.
+
+EXAMPLES
+    Automatically review Safe singleton factory deployment GitHub PR #42:
+        github-review.sh 42
+EOF
+}
+
+if [[ -f .env ]]; then
+    echo "ERROR: Please remove '.env' file as it interferes with this script" 1>&2
+    exit 1
+fi
+if [[ -n "$(git status --porcelain)" ]]; then
+    echo "ERROR: Dirty Git index, please commit all changes before continuing" 1>&2
+    exit 1
+fi
+if ! command -v gh &> /dev/null; then
+    echo "ERROR: Please install the 'gh' GitHub CLI" 1>&2
+    exit 1
+fi
+
+pr=0
+case $# in
+    1)
+        if ! [[ $1 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $1 is not a valid GitHub PR number" 1>&2
+            usage
+            exit 1
+        fi
+        pr=$1
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+esac
+
+echo "### Fetching RPC URL"
+issue="$(gh pr view $pr --json body --jq .body | sed -n 's/Fixes #\([0-9]\+\)/\1/p' | head -1)"
```

Comment:
```suggestion
issue=$(gh pr view $pr --json body --jq '.body | match("Fixes #[0-9]+") | .string | split(" ")[1] | sub("#"; "")' | head -1)
```

Can we make it like this? (It's ugly, I know 😞 But works, tried with #454 )

Reply:
It looks like a BSD vs GNU prob though.

Reply:
Nice - well at least you found the issue. I'm going to play around with BSD sed and see if I can express the same thing using POSIX subset of `sed` so it is portable.

Reply:
I was able to fix using `sed -E` for better regexp portability (which is supported by both GNU and BSD sed).

Reply:
Aaaaaand, I found a way with POSIX `sed`, so its ultraportable.

The issue is that `+` is not supported by the POSIX BRE (regular expressions) standard, but an extension in GNU.